contrib/node/tests/python/pants_test/contrib/node/tasks/test_node_resolve.py

```diff
@@ -230,3 +239,53 @@ def test_resolve_preserves_package_json(self):
       self.assertNotIn('devDependencies', package)
       self.assertNotIn('peerDependencies', package)
       self.assertNotIn('optionalDependencies', package)
+
+  def _test_resolve_optional_install_helper(
+    self, install_optional, expected_params, package_manager):
+    typ = self.make_target(spec='3rdparty/node:typ', target_type=NodeRemoteModule, version='0.6.3')
+    self.create_file('src/node/util/package.json', contents=dedent("""
+      {
+        "name": "util",
+        "version": "0.0.1"
+      }
+    """))
+    self.create_file('src/node/util/util.js', contents=dedent("""
+      var typ = require('typ');
+      console.log("type of boolean is: " + typ.BOOLEAN);
+    """))
+    # yarn execution path requires yarn.lock
+    self.create_file('src/node/util/yarn.lock', contents='')
+    target = self.make_target(spec='src/node/util',
+                              target_type=NodeModule,
+                              sources=['util.js', 'package.json', 'yarn.lock'],
+                              dependencies=[],
+                              package_manager=package_manager)
+
+    context = self.context(target_roots=[target], options={
+      'npm-resolver': {'install_optional': install_optional}
+    })
+    task = self.create_task(context)
+
+    method_to_mock = {
+      'npm': 'execute_npm',
+      'yarn': 'execute_yarnpkg'
+    }[package_manager]
+    with mock.patch.object(task, method_to_mock) as exec_call:
+      exec_call.return_value = (0, None)
+      task.execute()
+      exec_call.assert_called_once_with(
+        expected_params,
+        workunit_labels=mock.ANY,
+        workunit_name=mock.ANY)
+
+  def test_resolve_default_no_optional_install_npm(self):
+    self._test_resolve_optional_install_helper(False, ['install', '--no-optional'], 'npm')
+
+  def test_resolve_optional_install_npm(self):
+    self._test_resolve_optional_install_helper(True, ['install'], 'npm')
+
+  def test_resolve_default_no_optional_install_yarn(self):
+    self._test_resolve_optional_install_helper(False, ['--ignore-optional'], 'yarn')
+
+  def test_resolve_optional_install_yarn(self):
+    self._test_resolve_optional_install_helper(True, [], 'yarn')
```

Comment:
Could you add the kwarg names to the args here? Also, I'd prefer if the expected_params were the last arg since I think it reads better that way.

eg

```
    self._test_resolve_optional_install_helper(
      install_optional=False,
      package_manager='npm',
      expected_params=['install', '--no-optional']
    )
```

Reply:
done.